tests/Microsoft.Identity.Test.Common/TestCommon.cs

```diff
@@ -45,6 +46,7 @@ public static void ResetInternalStaticCaches()
             OidcRetrieverWithCache.ResetCacheForTest();
             AuthorityManager.ClearValidationCache();
             SingletonThrottlingManager.GetInstance().ResetCache();
+            ManagedIdentityClient.ResetManagedIdentitySourceCache();

```

Comment:
Doesn't the method `ResetInternalStaticCaches` need to be exposed in public API for Azure SDK ? Probably experimental extensiblity PR. Can be as a separate PR, just making sure it's on your radar.